internal/handshake/xor_nonce_aead_boring.go

```diff
@@ -11,20 +11,18 @@ import (
 )
 
 var (
-	useBoring             = len(os.Getenv("USE_BORING")) != 0
-	errBoringIsNotEnabled = errors.New("boring was requested but not enabled")
+	errBoringIsNotEnabled      = errors.New("boring is not enabled")
+	goBoringDisabled      bool = os.Getenv("QUIC_GO_DISABLE_BORING") == "1"
 )
 
 func newAEAD(aes cipher.Block) (cipher.AEAD, error) {
-	if useBoring {
-		if boring.Enabled() {
-			return tls.NewGCMTLS13(aes)
-		} else {
-			return nil, errBoringIsNotEnabled
-		}
-	} else {
+	if goBoringDisabled {
 		return cipher.NewGCM(aes)
 	}
+	if !boring.Enabled() {
```

Comment:
this if is impossible to happen right?

Reply:
No, you can use a vanilla Go compiler and set the buildtags to include the boringcrypto which will compile this source file however at runtime it will crash.

Reply:
So this is just a runtime assertion to make sure that we are using the boringcrypto compiler when building this source file?

Reply:
So to clarify.
My initial rationale was: this assertion makes sure that at runtime BoringCrypto is enabled otherwise calling NewGCMTLS13 would fail.
The above is wrong and as Oliveirinha questioned the if is indeed impossible.

What happens with the compiler during the compilation process is it looks that the tags and selects the appropriate files, in this case the boringcrypto is present hence this file is used, transitively boring.Enabled() is always true otherwise this file would never be included in the compilation process.

So @joliveirinha your are correct the boring.Enabled() is unnecessary here and below as we rely on the tags in the top of the file to select between internal/handshake/xor_nonce_aead_boring.go and internal/handshake/xor_nonce_aead_noboring.go